packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -279,15 +281,14 @@ function WorklistDataGrid(props) {
     setFocusedRowCol(newFocusCell.row, newFocusCell.col, false);
 
     if (!hasSelectableRows) {
-      rowSelection.current.rangeUsingFocusedRow = null;
-      rowSelection.current.rangeUsingLastIndividuallySelectedRow = null;
+      multiSelectRange.current = {};
     }
 
     // Since the row selection mode has changed, the row selection mode needs to be updated.
-    if (!rowSelection.current.isRowSelectionModeToggledByGrid) {
+    if (!isRowSelectionModeToggledByGrid.current) {
```

Comment:
Why would the row selection being toggled by the grid impact this message?  Aren't we still enabling or disabling row-selection mode?  It really seems like we should be able to get rid of this variable.

Reply:
Once we start deriving the messages after the consumer responds then this will go away. The requirements states that when the row selection is enabled or disabled from outside the grid, the grid also needs to announce the change in row selection state. That is currently being done on the re-render based on change to the hasSelectableRows prop. However, when row selection is enabled or disabled from within the grid, the message is immediately announced. Therefore on the re-render, we needed a way to determined if the change was triggered from inside or outside the grid so that the message was not announce multiple times.